lib/mint/http1/headers.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Mint.HTTP1.Headers do
+  alias Mint.Core.Util
+
+  # format of headers is [{header_name, canonical_header_name, value}]
+
+  def from_raw_headers(headers) do
+    for {name, value} <- headers, do: {name, Util.downcase_ascii(name), value}
```

Comment:
Let's not use a `for` comprehension here, because it will silently skip elements of `headers` that don't look like `{name, value}`.

```suggestion
    Enum.map(headers, fn {name, value} ->
      {name, Util.downcase_ascii(name), value}
    end)
```

(plus what @ericmj said)